rfc/0022-policy-lifecycle.md

```diff
@@ -0,0 +1,869 @@
+The original module will remain in the shared cache, enabling the Policy Server to recompile it when necessary, such as when the Policy Server version changes.
+This ensures compatibility when the Policy Server is updated, and the precompiled modules might no longer work with the new version.
+Example: `/cache/ghcr.io/kubewarden/policies/safe-labels/v1.0.0/optimized-kw-v1.28.1.wasm`.
+Where `v1.0.0` is the policy version and `v1.28.1` the policy-server version.
+
+#### Security Implications
+
+Loading precompiled binary blobs from a shared cache introduces potential security risks.
+An attacker could replace a valid binary blob with one that runs arbitrary malicious code.
+
+To mitigate this risk, we will use [sigstore](https://github.com/sigstore)'s signing capabilities to sign and verify all binary blobs.
+By leveraging Kubewarden's internal certificate authority (CA), we can generate a certificate and private key for the leader to sign blobs, ensuring that only trusted and verified blobs are used.
+
+### The Policy Server is a StatefulSet
+
+The Policy Server will be deployed as a StatefulSet to ensure that each replica has a unique identity.
+This will allow us to identify the status of a policy on a specific Policy Server replica.
+
+### Policy Status
+
+#### Status Conditions
+
+We will use the [status conditions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) convention to report the status of the policy.
+The status condition struct will be extended to include the replica number.
+The approach of extending the status condition struct is already used in the Kubernetes codebase,
+for example in the [Pod conditions](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions).
+
+The `PolicyStatusCondition` struct will have the following fields:
+
+| **Field name**         | **Description**                                                                                     |
+| ---------------------- | --------------------------------------------------------------------------------------------------- |
+| **type**               | Name of the policy condition.                                                                       |
+| **status**             | Indicates whether the condition is applicable, with possible values: "True", "False", or "Unknown". |
+| **reason**             | Machine-readable reason for the condition's last transition, in UpperCamelCase format.              |
+| **message**            | Human-readable message providing details about the last transition.                                 |
+| **generation**         | The generation of the `Policy` the status is referring to.                                          |
+| **replica** (optional) | The replica number reporting the status.                                                            |
```

Comment:
I'm struggling to see the status.replica real usefulness. Also, the replica number may point to a different policy-server in the future. I think it would be better to emit events when we change statuses instead as mentioned later.